File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
     
     
	int res = nums[0];
	int count = 1;

	for(int i = 1; i < (int)nums.size() ; i ++){
		if(nums[i] == res)
			count ++;
		else {
			count -=1;
			if(count == 0)
			{
				res = nums[i];
				count = 1;
			}
		}
	}

	return res;
   
        
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        count = 1
        res = nums[0]

        for i in range(1, len(nums)):
            if nums[i] == res:
                count += 1

            else:
                count -= 1
                if count <= 0:
                    res = nums[i]
                    count = 1

        return res
